server/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -790,6 +790,21 @@ public void onFailure(final Exception e) {
                 }
             }, TimeValue.timeValueSeconds(30));
             latch.await();
+
+            // It's possible to acquire permits if we skip the primary mode check
```

Comment:
I see above in the test that

```
indexShard.getReplicationTracker().isPrimaryMode() == isPrimaryMode;
```

So can we make these new tests on an index shard whose replication tracker is not in primary mode?

Reply:
There is `if (isPrimaryMode) {` check on the line 756 right after this assertion and we make these new tests in the `else` branch, so do we do test that we can acquire permits when the shard in a non-primary mode.

Reply:
Oh thanks, now saw it.